alpine-nix-rails-nginx/etc/nginx/dynamic-content.conf

```diff
@@ -0,0 +1,20 @@
+# Disable caching for dynamic content
+expires off;
+
+# Work-around for BREACH attack: http://breachattack.com/
+gzip off;
+
+# Restrict allowed methods
+if ($request_method !~ ^(DELETE|GET|HEAD|OPTIONS|PATCH|POST|PUT)$) {
+  more_set_headers 'Allow: DELETE,GET,HEAD,OPTIONS,PATCH,POST,PUT';
+  return 405;
+}
+
+# Handle OPTIONS requests
+if ($request_method = OPTIONS) {
```

Comment:
I could look and see if there's a way to only use `OPTIONS` for non-dynamic URLs. Then I could just have it return `GET,HEAD`; and then delegate any other `OPTIONS` handling to rails itself.